src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -124,9 +124,7 @@ public CSharpCodeBlockSyntax ParseBlock()
                 {
                     NextToken();
 
-                    // Unless changed, the block is a statement block
-                    AcceptWhile(IsSpacingToken(includeNewLines: true, includeComments: true));
-                    builder.Add(OutputTokensAsStatementLiteral());
```

Comment:
I was always classifying it as `CSharpStatementLiteral` early on. The change is to keep it around until we know what kind of transition we are dealing with and then classify it correctly.